tests/CMakeLists.txt

```diff
@@ -63,7 +68,7 @@ message("valgrind suppression file is located at " ${VALGRIND_SUPPRESSION})
 
 if (LINUX)
     set(TESTLIB_OPTS_C snowflakeclient ${CMOCKA_LIB} -Wl,--whole-archive curl ssl crypto -Wl,--no-whole-archive pthread)
-    set(TESTLIB_OPTS_CXX snowflakeclient ${CMOCKA_LIB} -Wl,--whole-archive curl ssl crypto pthread aws-cpp-sdk-core aws-cpp-sdk-s3 -Wl,--no-whole-archive)
+    set(TESTLIB_OPTS_CXX snowflakeclientcpp snowflakeclient ${CMOCKA_LIB} -Wl,--whole-archive curl ssl crypto pthread aws-cpp-sdk-core aws-cpp-sdk-s3 -Wl,--no-whole-archive)
```

Comment:
Basically we want to put both CPP and C code together into a single library file.
The header file locations could be separated like `snowflakecpp`, but we will revisit it later.